base/server/scripts/nuxwdog

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+
+# Authors:
+#     Dinesh Prasanth M K <dmoluguw@redhat.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the Lesser GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+#  along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright (C) 2018 Red Hat, Inc.
+# All rights reserved.
+
+
+"""
+Script that prompts user for password during server startup
+and store them on user's keyring
+"""
+
+import getopt
+import logging
+import os
+import subprocess
+import sys
+
+import pki.server as server
+
+from pki.keyring import Keyring
+
+logger = logging.getLogger(__name__)
+
+logging.basicConfig(format='%(levelname)s: %(message)s')
+
+tags = set()
+
+# Create a instance of Keyring
+keyring = Keyring()
+
+
+def split_entries(entry):
+    return entry.split(',')
+
+
+def print_help():
+    print('Usage: nuxwdog [OPTIONS]')
+    print()
+    print('      --clear                Clear values stored in keyring.')
+    print('      --help                 Show help message.')
```

Comment:
I'm not familiar with what the original /usr/bin/nuxwdog command does, but it doesn't seem to support these parameters. If we're introducing a new functionality (i.e. clearing the keyring), we might as well introduce a new command (e.g. pki-server keyring-clear) rather than adding a new parameter (i.e. --clear) to an old command.

Reply:
Also, I'm not sure if users are expected to call the old /usr/bin/nuxwdog directly. We need to check the documentation. If they aren't, we can replace it with a new command (e.g. pki-server keyring-update) and not worry about compatibility/conflicts.

Reply:
@edewata 

> I'm not familiar with what the original /usr/bin/nuxwdog command does, but it doesn't seem to support these parameters. If we're introducing a new functionality (i.e. clearing the keyring), we might as well introduce a new command (e.g. pki-server keyring-clear) rather than adding a new parameter (i.e. --clear) to an old command.

In original nuxwdog, the keyring gets cleared when the nuxwdog process exits (ie) the original nuxwdog used process keyring. Hence, there was no need for a specific clear option.

In this PR, `--clear` is specifically required since the new version uses `user keyring`. It's the duty of the caller to clear the keyring when the keys are no longer deemed to be useful. 

 




> Also, I'm not sure if users are expected to call the old /usr/bin/nuxwdog directly. We need to check the documentation. If they aren't, we can replace it with a new command (e.g. pki-server keyring-update) and not worry about compatibility/conflicts.

No. The users are supposed to interact with the /usr/bin/nuxwdog through systemd script. See this [original systemd script](https://github.com/dogtagpki/pki/blob/8c58112f0c672d45e4c1075ac5f79743320aabb7/base/server/share/lib/systemd/system/pki-tomcatd-nuxwdog%40.service#L14).